src/shared/sam/debugger/commands/addSamDebugConfiguration.ts

```diff
@@ -22,7 +25,40 @@ export async function addSamDebugConfiguration({
     // tslint:disable-next-line: no-floating-promises
     emitCommandTelemetry()
 
-    const samDebugConfig = createAwsSamDebugConfig(resourceName, templateUri.fsPath)
+    const workspaceFolder = vscode.workspace.getWorkspaceFolder(templateUri)
+    let preloadedConfig = undefined
+
+    if (workspaceFolder) {
+        const registry = CloudFormationTemplateRegistry.getRegistry()
+        const templateDatum = registry.getRegisteredTemplate(templateUri.fsPath)
+        if (templateDatum) {
+            const resource = templateDatum.template.Resources![resourceName]
+            if (resource && resource.Properties) {
+                const existingConfig = await getExistingConfiguration(
+                    workspaceFolder,
+                    resource.Properties.Handler,
+                    templateUri
+                )
+                if (existingConfig) {
+                    const responseYes: string = localize('AWS.generic.response.yes', 'Yes')
+                    const responseNo: string = localize('AWS.generic.response.no', 'No')
+                    const prompt = await vscode.window.showInformationMessage(
```

Comment:
The problem with the default notices is that they can be ignored. When the toolkit waits for a response, but the user didn't explicitly dismiss or respond to the notice, the user can think the system is non-responsive.

I suggest using a modal dialog in this case, otherwise a picker (less preferred) because you want to do something in direct response to a user gesture.

Reply:
I'll give it a look. The toast won't block the IDE/toolkit but can lead to a confusing interaction (specifically: the user doesn't see the toast, clicks it a few minutes later, and is wondering why the launch.json file popped up (the user might also not see the config if the file is long enough)).

The other thing I'm considering is adding a second codelens to trigger the debug config creation with old configuration loaded in. I'll try both and take some screenshots.

Reply:
Tested the modal: it works and it's less offensive than I thought it'd be, but it does come with a "Cancel" button by default that behaves the same as "No". Thoughts on how to handle this? Should I make "Cancel" or "X" kill the debug creation execution?

Reply:
Sounds like the message and buttons can articulate a clean user flow. Here's an idea...
Dialog text:

> An existing legacy configuration was detected for this function handler. Would you like to create a launch profile from scratch, or from the existing configuration?

And providing the following buttons:
* `New configuration` (makes an all new config)
* `Use existing configuration` (makes new config based on the legacy config)
* `Cancel` (backs out and does nothing)